proposal.rst

```diff
@@ -267,8 +292,69 @@ Project Timeline
 .. If you will have other commitments during the project time (for example,
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
-
-FILLME
+In this summer I would like to do three things:
+
+    1. Make the go bindings compatible with the xapian build system.
+
+    2.  Provide bindings in go and complete API reference (as it is provided for C++)
+        using godoc,Go lang's official documentation tool where any one can look at the 
+        documentation if they just have go installed.The user can view the xapian 
+        documentation by just running go doc xapian from the terminal or command prompt.
+        And the documentation would be providied in godoc official website 
+        and xapian.org.
+        I have divided this part of work into to 10 weeks.
+
+    3.  Provide the go get feature for using xapian bindings in go with 
+        proper guide lines regarding whether the xapian core is installed
+        or not.
+
+
+Before May 18,2020 :
+I would finish up the build system part of work for go bindings that 
+are generated purely from xapian-headers.i and xapian-head.i such that
+these can be used like any other bindings and in any platform that other xapian bindings can be used in.
+
+Each week for first 3 days I would work on provding API , that is writing swig interface 
+for the functions and classes which aren't generated from 
+existing swig interfaces(xapian-headers.i and xapian-head.i) with typemaps and other features of swig in go.i.
+Next 2 days documenting the code with examples and the last day for testing.
+I have grouped different classes for each week.
+
+
+First Month : 
+
+  Week1 :
+        Work on Document , Registry , Query , Stem implementation , Stem classes.
```

Comment:
You've listed a bunch of classes in each week, but it's not clear what you intend to do for them. Generally there are going to be several things you need to do for each class:

1. Wrapping anything that SWIG doesn't do automatically. This may involve typemaps, or may involve more complex custom code and/or additional SWIG directives.
2. Making the resultant API more idiomatic for golang users. Typically the main wins are in iterators, but there may be other things that better match go if they are abstracted further away from the C++ API.
3. Making sure exceptions work as intended. golang doesn't have exceptions; instead it has a return code pattern.
4. Supporting Directors for the class if necessary (which is a SWIG concept allowing you to subclass a C++ class in the target language). This only applies to some classes; most of Xapian you don't need or want to subclass. (Document, for instance, isn't helpful to subclass. MatchSpy is.)

What I think you should do as much as possible this week is to look through the different classes you've listed, and figure out which of the things above make sense to do. Then you may want to think in terms of functionality to deliver piece by piece. For instance, you might want to first get to a place where you can write a simple query system in golang is probably helpful. For that, you need at least Database, Document, Enquire, Query, QueryParser, MSet and MSetIterator, and probably Stem (but you're unlikely to need StemImplementation) — possibly with some methods ignored. Then you could add basic indexing (WritableDatabase and TermGenerator), and then gradually fill out the other functionality (which will depend on what you think is most useful — perhaps you want to add support for Stopper, or for the other iterators, or for configuring the weighting scheme used when querying).

A good way of figuring out the chunks of functionality would be to run through the getting started / user guide, which introduces different classes and functionality in a sensible order.

Reply:
There's probably nothing needed over the default SWIG wrapping for some classes.  For example, there's no special handling for `Xapian::Registry` in any of the existing bindings, so I doubt any would be needed for Go (try `git grep -w Registry` in xapian-bindings to see this).